odoo/tests/common.py

```diff
@@ -1636,7 +1642,13 @@ def browser_js(self, url_path, code, ready='', login=None, timeout=60, cookies=N
             timeout = timeout * 1.5
 
         self.start_browser()
-
+        if watch and self.browser._webkit_version:
+            _logger.warning('watch mode is only suitable for local testing')
+            version_hash = self.browser._webkit_version.split('@')[-1].strip((')'))
+            chrome_dev_tool_url = 'https://chrome-devtools-frontend.appspot.com/serve_file/@%s/inspector.html' % version_hash
+            debug_url = '%s?%s&remoteFrontend=true' %(chrome_dev_tool_url, self.browser.ws_url.replace('://', '='))
```

Comment:
Surely it's not necessary to have *internet access* to a random appspot app to connect a local chrome instance to an other local chrome instance?

The accepted answer to https://stackoverflow.com/questions/63243137/devtools-remote-debugging-at-chrome-not-working-normally-with-headless-chrome seems to indicate we can just connect to the `ws_url` over http (rather than `ws://`) to get a selector of acceptable pages, and ` /devtools/inspector.html` can be given a page to inspect directly.

Reply:
I have no clue why I did this initially, I may have followed the wrong path

We found a cleaner solution with @d-fence